projects/vdk-core/tests/vdk/internal/builtin_plugins/ingestion/test_ingester_base.py

```diff
@@ -14,7 +15,7 @@
 
 shared_test_values = {
     "test_payload1": {"key1": "val1", "key2": "val2", "key3": "val3"},
-    "test_payload2": {"key1": 1, "key2": 2, "key3": 3},
+    "test_payload2": {"key1": Decimal(1), "key2": 2, "key3": 3},
```

Comment:
If you revert your change, will the modified test fail? This is sure way to 
A) check that the test is indeed verifying the bug you have. 
B) that you have really fixed the bug properly

I suspect it should fail in this case but let's check.

Reply:
I've checked this, it does indeed fail.